### PR TITLE
release trtl iterators

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This fixes a bug in the trtl store tests where we were failing to close iterators properly after using them, which manifested in race conditions during test suite teardown which were difficult to track down.

SC-6488

### Type of change

- [x] bug fix
- [ ] new feature
- [ ] documentation
- [ ] other (describe)

### Acceptance criteria

The race conditions in the tests should be addressed, unfortunately I was never able to reproduce them locally but so far they haven't shown up in CI yet 🤞 .

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] For every spot in the tests where we are using the raw trtl streaming iterator, we are calling `Release()` to close the iterator properly on the client side.

### CODE REVIEW

File: pkg/gds/store/trtl/trtl_test.go [hunk 1]
```diff
@@ -223,6 +224,7 @@ func (s *trtlStoreTestSuite) TestDirectoryStore() {
 	require.NoError(err)
```
Comment (bbengfort) [inline pkg/gds/store/trtl/trtl_test.go:224]: On [Line 202](https://github.com/trisacrypto/directory/blob/e0be55fbb8477f63a905d6aa34079fd9f170dbd3/pkg/gds/store/trtl/trtl_test.go#L202) above and [Line 245](https://github.com/trisacrypto/directory/blob/e0be55fbb8477f63a905d6aa34079fd9f170dbd3/pkg/gds/store/trtl/trtl_test.go#L245) below, does calling `All()` release the iterator?

**Review by bbengfort — APPROVED** [review]
Thank you for figuring this out - it's definitely been annoying to have CI failing on us and this was definitely a hidden issue; nice one!

File: pkg/gds/store/trtl/trtl_test.go (hunk 1 above)
Comment (pdeziel) [reply pkg/gds/store/trtl/trtl_test.go:224]: Yes, `ListVASPs()` creates the iterator, and `All()` releases the iterator when it returns.

Comment (bbengfort) [reply pkg/gds/store/trtl/trtl_test.go:224]: Cool - then this looks good to go to me! Hope it sorts out CI!

**Pull request merged**